packages/eslint-plugin/src/rules/space-infix-ops.ts

```diff
@@ -119,11 +128,38 @@ export default util.createRule<Options, MessageIds>({
       checkAndReportAssignmentSpace(node, leftNode, rightNode);
     }
 
+    /**
+     * Check if it is missing spaces between type annotations chaining
+     * @param typeAnnotation TypeAnnotations list
+     */
+    function checkForTypeAnnotationSpace(
+      typeAnnotation: TSESTree.TSIntersectionType | TSESTree.TSUnionType,
+    ): void {
+      const UNIONS = ['|', '&'];
+      const types = typeAnnotation.types;
+
+      types.forEach(type => {
+        const operator = sourceCode.getTokenBefore(type);
+
+        if (!!operator && UNIONS.includes(operator.value)) {
+          const prev = sourceCode.getTokenBefore(operator);
+          const next = sourceCode.getTokenAfter(operator);
+
+          if (
+            !sourceCode.isSpaceBetweenTokens(prev!, operator) ||
+            !sourceCode.isSpaceBetweenTokens(operator, next!)
+          ) {
+            report(operator, operator);
```

Comment:
should this be reporting on the type node instead of the operator?

```suggestion
            report(typeAnnotation, operator);
```

Reply:
Tests still passes when chaging it. Make more sens so i changed it for typeAnnotation